### PR TITLE
[PW-3258] Apple pay [PW-3321] Google pay

### PR DESCRIPTION
**Pull request by cyattilakiss**
## Summary
Add support for Apple pay and Google pay (and in general prepare wallet payment methods)
In 1.7 toggle the Google pay button depending on the place order button's availability 
In 1.6 place order button only depends on the payment method itself therefore the toggle is not applicable

## Tested scenarios
Apple pay in iphone simulator in prestashop 1.6 and 1.7
Google pay in 1.6 and 1.7

### CODE REVIEW

**Review by msilvagarcia — DISMISSED** [review]

File: adyenofficial.php [hunk 1]
```diff
@@ -816,6 +838,70 @@ public function displayForm()
...
+            'submit' => array(
+                'title' => $this->l('Save'),
+                'class' => 'btn btn-default pull-right'
+            )
+        );
+
+        // Apple pay merchant name input
+        $fields_form[1]['form']['input'][] = array(
+            'type' => 'text',
+            'label' => $this->l('Apple pay merchant name'),
```
Comment (acampos1916) [inline adyenofficial.php:856]: Just some caps on the labels
```suggestion
            'label' => $this->l('Apple Pay merchant name'),
```

File: adyenofficial.php [hunk 2]
```diff
@@ -816,6 +838,70 @@ public function displayForm()
...
+            'lang' => false,
+            'hint' => $this->l(
+                ''
+            )
+        );
+
+        // Apple pay merchant identifier input
+        $fields_form[1]['form']['input'][] = array(
+            'type' => 'text',
+            'label' => $this->l('Apple pay merchant identifier'),
```
Comment (acampos1916) [inline adyenofficial.php:869]:
```suggestion
            'label' => $this->l('Apple Pay merchant identifier'),
```

File: adyenofficial.php [hunk 3]
```diff
@@ -816,6 +838,70 @@ public function displayForm()
...
+            'lang' => false,
+            'hint' => $this->l(
+                ''
+            )
+        );
+
+        // Google pay gateway merchant id
+        $fields_form[1]['form']['input'][] = array(
+            'type' => 'text',
+            'label' => $this->l('Google pay gateway merchant id'),
```
Comment (acampos1916) [inline adyenofficial.php:882]:
```suggestion
            'label' => $this->l('Google Pay gateway merchant ID'),
```

File: adyenofficial.php [hunk 4]
```diff
@@ -816,6 +838,70 @@ public function displayForm()
...
+            'lang' => false,
+            'hint' => $this->l(
+                ''
+            )
+        );
+
+        // Google pay merchant identifier input
+        $fields_form[1]['form']['input'][] = array(
+            'type' => 'text',
+            'label' => $this->l('Google pay merchant identifier'),
```
Comment (acampos1916) [inline adyenofficial.php:895]:
```suggestion
            'label' => $this->l('Google Pay merchant identifier'),
```

File: views/js/checkout-component-renderer.js [hunk 5]
```diff
@@ -67,351 +75,525 @@ jQuery(document).ready(function () {
...
-        componentPersonalDetails = {
-            firstName: prestashop.customer.firstname,
-            lastName: prestashop.customer.lastname,
-            shopperEmail: prestashop.customer.email,
-            telephoneNumber: phoneNumber,
-            gender: getAdyenGenderByPrestashopType(prestashop.customer.gender.type),
-            dateOfBirth: prestashop.customer.birthday
-        }
-    }
+        // use this object to iterate through the stored payment methods
```
Comment (acampos1916) [inline views/js/checkout-component-renderer.js:188]: Are these stored payment methods? Maybe it is a left over
```suggestion
        // use this object to iterate through the payment methods
```

**Review by acampos1916 — DISMISSED** [review]

File: views/js/checkout-component-renderer.js (hunk 5 above)
Comment (cyattilakiss) [reply views/js/checkout-component-renderer.js:188]: correct!

**Review by acampos1916 — APPROVED** [review]

**Review by msilvagarcia — APPROVED** [review]

**Pull request merged**